packages/flutter/lib/src/material/notches.dart

```diff
@@ -0,0 +1,174 @@
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+/// Computes notches in the outline of a shape.
+///
+/// Typically used to compute notches by a [BottomAppBar].
+abstract class Notch {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const Notch();
+
+  /// Returns a path for a notch in the outline of a shape.
+  ///
+  /// The path makes a notch in the host shape that can contain the guest shape.
```

Comment:
If you update the previous line, you could leave this one out.

Reply:
done